File: app/src/main/java/com/blundell/tut/MainActivity.java
package com.blundell.tut;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.ListView;

import java.util.List;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ListView listWidget = (ListView) findViewById(R.id.main_comments);
        CommentAdapter adapter = new CommentAdapter(getLayoutInflater(), new TimeStampFormatter());
        List<Comment> comments = MockBackend.loadComments();
        comments.addAll(MockBackend.loadIsoComments());
        adapter.updateWith(comments);
        listWidget.setAdapter(adapter);
    }

}

File: app/src/main/java/com/blundell/tut/MainActivity.kt
package com.blundell.tut

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.widget.ListView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val adapter = CommentAdapter(layoutInflater, TimeStampFormatter())
        val comments: ArrayList<Comment> = MockBackend.loadComments()
        comments.addAll(MockBackend.loadIsoComments())
        adapter.updateWith(comments)
        mainComments.adapter = adapter
    }

}
